src/data/ACTIONS/root/SGE.ts

```diff
@@ -1,5 +1,298 @@
+import {Attribute} from 'event'
 import {ensureActions} from '../type'
 
+const SGE_STUB = 190705000
+
 export const SGE = ensureActions({
-	// TODO: Add actions here
+	/** Single-Target Heals/Shields */
+	DIAGNOSIS: {
+		id: SGE_STUB + 1,
+		name: 'Diagnosis',
+		icon: '',
+		onGcd: true,
+		speedAttribute: Attribute.SPELL_SPEED,
+		castTime: 1500,
+		mpCost: 400,
+	},
+	EUKRASIAN_DIAGNOSIS: {
+		id: SGE_STUB + 2,
+		name: 'Eukrasian Diagnosis',
+		icon: '',
+		onGcd: true,
+		gcdRecast: 1500,
+		mpCost: 1000,
+		statusesApplied: ['EUKRASIAN_DIAGNOSIS', 'DIFFERENTIAL_DIAGNOSIS'],
+	},
+	HAIMA: {
+		id: SGE_STUB + 3,
+		name: 'Haima',
+		icon: '',
+		cooldown: 120000,
+		statusesApplied: ['HAIMATINON'],
```

Comment:
NBD right now but I'm assuming that Haimatinon will be a distinct buff from the actual shield, so once we have the name of that this'll probably want updating.

Reply:
Sure enough, there's both a `Haima` and a `Haimatinon` buff getting applied in some footage I checked. Added/fixed.